File: frontend/src/api/score.ts
import request from './client';

export interface ScoreDetail {
  id: number;
  event_name: string;
  group_name: string;
  gender: string;
  is_team_event: boolean;
  athlete_name: string | null;
  rank: number;
  result: string;
  base_score: number;
  is_record_broken: boolean;
  final_score: number;
  created_at: string;
}

export interface ClassScore {
  id: number;
  class_team_id: number;
  grade: string;
  class_name: string;
  total_score: number;
  rank_in_grade: number;
  updated_at: string;
}

export interface AthleteScore {
  id: number;
  athlete_id: number;
  class_team_id: number;
  grade: string;
  class_name: string;
  athlete_name: string;
  total_score: number;
  rank_in_grade: number;
  updated_at: string;
}

export interface CalculateSummary {
  total_details: number;
  total_athletes: number;
  total_classes: number;
  calculated_at: string;
}

/**
 * 一键统计
 */
export function calculateScores() {
  return request.post<CalculateSummary>('/scores/calculate').then(res => res.data);
}

/**
 * 获取班级得分列表
 */
export function listClassScores(grade?: string) {
  return request.get<ClassScore[]>('/scores/classes', { params: { grade } }).then(res => res.data);
}

/**
 * 获取班级得分明细
 */
export function getClassDetails(classTeamId: number) {
  return request.get<ScoreDetail[]>(`/scores/classes/${classTeamId}/details`).then(res => res.data);
}

/**
 * 获取个人得分列表
 */
export function listAthleteScores(grade?: string) {
  return request.get<AthleteScore[]>('/scores/athletes', { params: { grade } }).then(res => res.data);
}

/**
 * 获取个人得分明细
 */
export function getAthleteDetails(athleteId: number) {
  return request.get<ScoreDetail[]>(`/scores/athletes/${athleteId}/details`).then(res => res.data);
}

File: backend/app/schemas/score.py
from datetime import datetime

from pydantic import BaseModel


class ScoreDetailOut(BaseModel):
    """得分明细输出"""

    id: int
    event_name: str
    group_name: str
    gender: str
    is_team_event: bool
    athlete_name: str | None  # 个人项目才有
    rank: int
    result: str
    base_score: int
    is_record_broken: bool
    final_score: int
    created_at: datetime


class ClassScoreOut(BaseModel):
    """班级得分输出"""

    id: int
    class_team_id: int
    grade: str
    class_name: str
    total_score: int
    rank_in_grade: int
    updated_at: datetime


class AthleteScoreOut(BaseModel):
    """个人得分输出"""

    id: int
    athlete_id: int
    class_team_id: int
    grade: str
    class_name: str  # 额外字段，前端显示用
    athlete_name: str
    total_score: int
    rank_in_grade: int
    updated_at: datetime


class CalculateSummary(BaseModel):
    """统计计算摘要"""

    total_details: int  # 总得分明细数
    total_athletes: int  # 参与得分的运动员数
    total_classes: int  # 参与得分的班级数
    calculated_at: datetime
